packages/@pollyjs/adapter-xhr/src/index.js

```diff
@@ -61,10 +61,22 @@ export default class XHRAdapter extends Adapter {
     this.xhr.restore();
   }
 
+  onRequest(pollyRequest) {
+    const { xhr } = pollyRequest.requestArguments;
+
+    if (xhr.aborted) {
+      pollyRequest.abort();
+    } else {
+      xhr.addEventListener('abort', () => pollyRequest.abort());
```

Comment:
General comment that applies to these abort event listeners, we should clean these event listeners up when the request is complete or aborted.